lib/paypal_express/api.rb

```diff
@@ -84,37 +85,32 @@ def refund_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_p
       end
 
       def get_payment_info(kb_account_id, kb_payment_id, properties, context)
-        ignored_api_calls = [:details_for]
-        responses = @response_model.from_kb_payment_id(@transaction_model, kb_payment_id, context.tenant_id)
-        responses = responses.reject do |response|
-          ignored_api_calls.include?(response.api_call.to_sym)
-        end
-        t_info_plugins = responses.collect do |response|
-          response.to_transaction_info_plugin(response.send("#{@identifier}_transaction"))
-        end
+        filtered_plugin_info, plugin_info, with_only_pending_trx = get_raw_payment_info(kb_payment_id, context)
         # Should never happen...
-        return [] if t_info_plugins.nil?
+        return [] if filtered_plugin_info.nil?
 
-        # Completed purchases/authorizations will have two rows in the responses table (one for api_call 'build_form_descriptor', one for api_call 'purchase/authorize')
-        # Other transaction types don't support the :PENDING state
-        target_transaction_types = [:PURCHASE, :AUTHORIZE]
-        only_pending_transaction = t_info_plugins.find { |t_info_plugin| target_transaction_types.include?(t_info_plugin.transaction_type) && t_info_plugin.status != :PENDING }.nil?
-        t_info_plugins_without_pending = t_info_plugins.reject { |t_info_plugin| target_transaction_types.include?(t_info_plugin.transaction_type) && t_info_plugin.status == :PENDING }
+        options = properties_to_hash(properties)
+        # We won't be in a state where we have both a pending and unknown plugin infos; so we just return here.
+        if fix_unknown_transactions(kb_payment_id, plugin_info, options, kb_account_id, context) ||
+           cancel_pending_transactions(filtered_plugin_info, with_only_pending_trx)
```

Comment:
Yes. I think we won't be in a situation where we have both an unknown and a pending payment.